crypto/cryptlib.c

```diff
@@ -401,8 +401,11 @@ void *OPENSSL_stderr(void)	{ return stderr; }
 int CRYPTO_memcmp(const void *in_a, const void *in_b, size_t len)
 	{
 	size_t i;
-	const unsigned char *a = in_a;
-	const unsigned char *b = in_b;
+	/* Must use "pointers to volatile" to tell the compiler to never
+	 * attempt optimizing the reads. Otherwise the compiler is allowed
+	 * to use LTO and convert this function into canonical version. */
+	const volatile unsigned char *a = in_a;
```

Comment:
1. This makes code larger and much slower.
2. It doesn't actually make it any better. If memory pointed to wasn't volatile originally then it isn't any more volatile with this pointer declaration. The compiler is required to preserve the sequence of reads and writes for the pointer itself but not for the pointed to memory.